src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -362,46 +331,35 @@ internal PSKeyword GetLogKeywords()
             return result;
         }
 #endif // UNIX
-
-        private T ReadValueFromFile<T>(string fileName, string key)
+ 
+        private T ReadValueFromFile<T>(string fileName, string key, T defaultValue = default(T),
+                                       Func<JToken, JsonSerializer, T, T> readImpl = null)
         {
+            if (!File.Exists(fileName)) { return defaultValue; }
+
+            // Open file for reading, but allow multiple readers
             fileLock.EnterReadLock();
             try
             {
-                // Open file for reading, but allow multiple readers
-                using (FileStream fs = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.Read))
-                using (StreamReader streamRdr = new StreamReader(fs))
-                using (JsonTextReader jsonReader = new JsonTextReader(streamRdr))
+                using (var streamReader = new StreamReader(fileName))
+                using (var jsonReader = new JsonTextReader(streamReader))
                 {
-                    // Safely determines whether there is content to read from the file
-                    bool isReadSuccess = jsonReader.Read();
-                    if (isReadSuccess)
+                    var settings = new JsonSerializerSettings() { TypeNameHandling = TypeNameHandling.None, MaxDepth = 10 };
+                    var serializer = JsonSerializer.Create(settings);
+
+                    var configData = serializer.Deserialize<JObject>(jsonReader);
```

Comment:
Has a security review been done on this code?  Is it possible for the deserializer to create arbitrary types or run arbitrary code?  If so is it possible to apply a schema to mitigate?  This could be a possible security hole when we (re) implement PowerShell white listing protection.  Can you add a comment about this possible vulnerability and create an issue to track?

Reply:
> is it possible to apply a schema to mitigate

We have a PR for Test-Json - it can help here.

Reply:
@PaulHigin the use of `JsonSerailizer` here is the same as in `ConverFrom/ConverTo-Json` cmdlets. I will add a comment and create an issue.

Reply:
I had a closer look at the usage, and think it should be safe. The usage here can only convert values to the limited types: string, bool, string[], bool?, and sub types of `PolicyBase`.

Reply:
@daxian-dbw  If that is the case then we should be Ok.  Thanks.

Reply:
Do PowerShell silently eat wrong config options without the schema check? It is bad UX.

Reply:
PowerShell currently doesn't do schema check on the config file, which needs to be addressed. I opened #5809 with 3 tasks. Feel free to bring up more to discuss.